beacon-chain/cache/depositsnapshot/deposit_inserter.go

```diff
@@ -89,6 +92,10 @@ func (c *Cache) InsertFinalizedDeposits(ctx context.Context, eth1DepositIndex in
 	c.depositsLock.Lock()
 	defer c.depositsLock.Unlock()
 
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
```

Comment:
And here you ignore the context coming from `StartSpan` and do an error check on the original context.

Reply:
In practice this is fine because the cancellation won't come from tracing, it will come from the parent context, but in principle we should always use the deepest context possible.

Reply:
oh lol i expanded the code pane after seeing your slack message - I too missed the interleaved func def 🤦